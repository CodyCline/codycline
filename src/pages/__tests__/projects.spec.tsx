import React from "react"
import renderer from "react-test-renderer";
import { getPageQueryData } from 'gatsby-plugin-testing';
import Projects from "../projects";

describe("Projects", () => {
    it("renders GraphQL data correctly", async() => {
        const data = await getPageQueryData(`projects`);
        const tree = renderer.create(<Projects data={data} />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})