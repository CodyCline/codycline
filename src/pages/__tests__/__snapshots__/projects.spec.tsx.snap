// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects renders GraphQL data correctly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      className="nav__item home__icon"
      href="/"
    >
      <img
        alt="logo.png"
        src="test-file-stub"
        style={
          Object {
            "height": "50px",
            "width": "70px",
          }
        }
      />
    </a>
    <a
      className="nav__item"
      href="/blog"
    >
       Blog 
    </a>
    <a
      className="nav__item"
      href="/snippets"
    >
      Snippets
    </a>
    <a
      className="nav__item"
      href="/projects"
    >
      Projects
    </a>
    <a
      className="nav__item controls"
      href="#"
    />
  </nav>,
  <div
    className="content__container"
  >
    <div
      style={
        Object {
          "height": "10vh",
          "width": "undefinedvh",
        }
      }
    />
    <div
      className="content__header"
    >
      <h2
        className="no__margin"
      >
        Projects
      </h2>
      <p>
        Projects, mostly-web based and open source.
      </p>
      <hr
        className="divider"
      />
    </div>
    <div
      className="content__card-grid"
    >
      <div
        className="card"
      >
        <div
          className="card__icon"
        >
          <svg
            className="card__icon-platform"
            height={72}
            viewBox="0 0 1024 1024"
            width={72}
          >
            <title>
              package
            </title>
            <path
              d="M551.055 192l367.357 97.92 41.6 30.080v431.36l-23.68 30.72-392.317 108.16-392.96-108.16-23.040-30.72v-431.36l39.040-30.080 366.72-97.92h17.28zM545.295 256l-256 64 35.2 12.8 219.52 57.6 191.997-51.84 60.8-18.56-251.517-64zM192.015 727.040l320 87.68v-366.72l-320-85.76v364.8zM576.015 448v366.72l319.997-87.68v-366.72l-129.274 35.423v164.258l-64 16.64v-163.362l-126.724 34.721z"
              fill={null}
              opacity={1}
            />
          </svg>
        </div>
        <ul
          className="card__meta card__links"
        >
          <li
            className="card__links-item"
          >
            <div
              className="tooltip__wrapper"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                href="https://github.com/codycline/paypal-react.git"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  height={28}
                  viewBox="0 0 1024 1024"
                  width={28}
                >
                  <path
                    d="M943.104 472.096l-391.2-391.2c-10.431-10.466-24.859-16.941-40.8-16.941s-30.369 6.476-40.798 16.94l-0.002 0.002-81.248 81.248 103.040 103.040c6.582-2.315 14.171-3.653 22.072-3.653 37.838 0 68.512 30.674 68.512 68.512 0 8.061-1.392 15.796-3.949 22.98l0.149-0.479 99.2 99.2c6.642-2.358 14.302-3.72 22.28-3.72 37.873 0 68.576 30.703 68.576 68.576s-30.703 68.576-68.576 68.576c-37.873 0-68.576-30.703-68.576-68.576 0-9.428 1.902-18.411 5.344-26.587l-0.169 0.451-92.64-92.64v243.744c22.757 11.458 38.095 34.633 38.095 61.39 0 37.838-30.674 68.512-68.512 68.512s-68.512-30.674-68.512-68.512c0-28.45 17.341-52.849 42.029-63.206l0.452-0.168v-245.984c-25.212-10.543-42.606-35.007-42.606-63.533 0-9.571 1.958-18.685 5.496-26.963l-0.17 0.448-101.44-101.504-268.256 268.352c-10.475 10.432-16.957 24.867-16.957 40.816s6.482 30.384 16.956 40.814l0.002 0.002 391.2 391.072c10.431 10.466 24.859 16.941 40.8 16.941s30.369-6.476 40.798-16.94l0.002-0.002 389.408-389.504c10.475-10.432 16.957-24.867 16.957-40.816s-6.482-30.384-16.956-40.814l-0.002-0.002z"
                    fill="rgb(221, 76, 53)"
                    opacity={1}
                  />
                  <path
                    d="M389.056 162.144l103.040 103.040c6.582-2.315 14.171-3.653 22.072-3.653 37.838 0 68.512 30.674 68.512 68.512 0 8.061-1.392 15.796-3.949 22.98l0.149-0.479 99.2 99.2c6.642-2.358 14.302-3.72 22.28-3.72 37.873 0 68.576 30.703 68.576 68.576s-30.703 68.576-68.576 68.576c-37.873 0-68.576-30.703-68.576-68.576 0-9.428 1.902-18.411 5.344-26.587l-0.169 0.451-92.64-92.64v243.744c22.757 11.458 38.095 34.633 38.095 61.39 0 37.838-30.674 68.512-68.512 68.512s-68.512-30.674-68.512-68.512c0-28.45 17.341-52.849 42.029-63.206l0.452-0.168v-245.984c-25.212-10.543-42.606-35.007-42.606-63.533 0-9.571 1.958-18.685 5.496-26.963l-0.17 0.448-101.44-101.504z"
                    fill="rgb(255, 255, 255)"
                    opacity={1}
                  />
                </svg>
              </a>
            </div>
          </li>
          <li
            className="card__links-item"
          >
            <div
              className="tooltip__wrapper"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <a
                href="https://www.npmjs.com/package/paypal-button-react"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  height={28}
                  viewBox="0 0 1024 1024"
                  width={28}
                >
                  <path
                    d="M281.615 192h197.44c57.711 0 113.059 22.925 153.867 63.734 40.806 40.808 63.731 96.155 63.731 153.866v13.12c-0.083 57.656-23.046 112.921-63.844 153.66-40.799 40.739-96.098 63.62-153.754 63.62h-31.040v-64h31.040c40.572-0.084 79.462-16.216 108.18-44.875s44.931-67.514 45.1-108.085v-13.44c0.001-40.627-16.094-79.599-44.762-108.386s-67.572-45.044-108.198-45.214h-197.76c-40.737 0-79.805 16.183-108.611 44.988-14.263 14.263-25.578 31.196-33.297 49.832s-11.692 38.609-11.692 58.78v13.12c0.097 36.194 13.002 71.185 36.431 98.773s55.868 45.99 91.569 51.947v64c-52.62-6.232-101.158-31.451-136.508-70.924s-55.081-90.49-55.492-143.476v-13.44c0.085-57.685 23.037-112.984 63.827-153.773s96.088-63.743 153.773-63.827zM768.026 450.56v-64c52.73 6.246 101.35 31.556 136.717 71.164 35.36 39.608 55.027 90.779 55.283 143.876v13.12c-0.173 57.574-23.117 112.742-63.827 153.453s-95.878 63.661-153.453 63.827h-197.763c-57.545-0.083-112.714-22.957-153.434-63.622-40.721-40.659-63.677-95.795-63.846-153.338v-13.44c0-57.656 22.882-112.955 63.62-153.754s96.004-63.762 153.66-63.846h31.040v64h-31.040c-40.682 0.084-79.668 16.305-108.404 45.101s-44.876 67.817-44.876 108.499v13.12c0.084 40.627 16.261 79.565 44.988 108.294 28.727 28.723 67.665 44.902 108.292 44.986h197.763c40.57-0.083 79.456-16.218 108.179-44.877 28.717-28.659 44.928-67.514 45.101-108.083v-13.44c-0.026-36.249-12.896-71.316-36.333-98.971-23.43-27.654-55.917-46.104-91.667-52.069z"
                    fill={null}
                    opacity={1}
                  />
                </svg>
              </a>
            </div>
          </li>
        </ul>
        <div
          className="card__body"
        >
          <h3>
            <a
              href="/projects/react-paypal/"
            >
              React PayPal
            </a>
          </h3>
          <p>
            <a
              href="/projects/react-paypal/"
            >
              JavaScript library of PayPal smart buttons for React
            </a>
          </p>
        </div>
        <ul
          className="card__meta card__status"
        >
          <li
            className="card__status-item"
          />
          <li
            className="card__status-item"
          />
        </ul>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "height": "60vh",
        "width": "undefinedvh",
      }
    }
  />,
  <footer
    className="footer"
  >
    <li
      className="footer__item"
    >
      2021
    </li>
    <li
      className="footer__item"
    />
  </footer>,
]
`;
